src/main/java/de/tum/in/www1/artemis/domain/iris/IrisSessionExerciseConnector.java

```diff
@@ -0,0 +1,60 @@
+package de.tum.in.www1.artemis.domain.iris;
+
+import java.time.ZonedDateTime;
+
+import javax.persistence.*;
+
+import org.hibernate.annotations.Cache;
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import de.tum.in.www1.artemis.domain.DomainObject;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.iris.session.IrisSession;
+
+/**
+ * An IrisSessionExerciseConnector is used if one student has multiple IrisSessions for one exercise
+ */
+@Entity
+@Table(name = "iris_session_exercise_connector")
+@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class IrisSessionExerciseConnector extends DomainObject {
```

Comment:
Why do you need this class? IrisChatSession already has a connection to a ProgrammingExercise, so this class is redundant.
Also, it is currently directly referencing IrisSession, but this does not make sense, as IrisHestiaSession is not connected to a ProgrammingExercise, but to a CodeHint.

And the creationDate could just directly be added to IrisSession